include/pybind11/cast.h

```diff
@@ -377,6 +377,22 @@ template <typename StringType, bool IsView = false> struct string_caster {
 #endif
         }
 
+#if PY_VERSION_HEX >= 0x03030000
+        // On Python >= 3.3, for UTF-8 we avoid the need for a temporary `bytes`
+        // object by using `PyUnicode_AsUTF8AndSize`.
+        if (PYBIND11_SILENCE_MSVC_C4127(UTF_N == 8)) {
+            Py_ssize_t size; // NOLINT(cppcoreguidelines-init-variables)
```

Comment:
Initializing this is cheap and prevents potential UB. I would recommend setting it to obvious error vals such as size=0 or size=-1. That way it makes it more obvious when an error occurs.

Reply:
In this case `size` is an output-only parameter of `PyUnicode_AsUT8AndSize`.  The idea of initializing it would be to protect against a bug in `PyUnicode_AsUT8AndSize`?

In general I don't agree with adding extra initializations as it makes it less obvious what is happening, and prevents sanitizers like memorysanitizer from detecting bugs.

However if the extra initialization is the preferred style in this code base I am happy to add it

Reply:
I agree with this. Looks much nicer, surely correct, runtime overhead certainly in the noise, if any.

Reply:
Yes, the extra initialization is the preferred style in this code base and the runtime overhead is minimal (if the compiler doesn't eliminate the unused_value entirely).

Reply:
@rwgk To clarify, who are you agreeing with?

Reply:
Sorry: I agree with setting a value, `Py_ssize_t size = -1;` preferred.

Reply:
Done.